server/controllers/uploadImage.js

```diff
@@ -0,0 +1,21 @@
+const asyncHandler = require("express-async-handler");
+const { uploadFile } = require('../utils/s3');
+const fs = require('fs');
+const util = require('util');
+const unlinkFile = util.promisify(fs.unlink);
+
+
+// @route POST /uploadImage
+// @desc Upload image to S3
+// @access Private
+exports.uploadImage = asyncHandler(async (req, res, next) => {
+  try {
+    const file = req.file;
+    const result = await uploadFile(file);
+    await unlinkFile(file.path);
+    res.status(200).json({ imagePath: `/images/${result.Key}`});
+  }
+  catch (err) {
+    res.status(500).json({ error: err });
+  }
```

Comment:
This is nice 👍🏼

We don't actually need to wrap this in `try...catch` since `asyncHandler` (i.e. `express-async-handler` package) forwards the error to the next handler for us (the way our app is wired, this turns out to be the `error` middleware which does exactly what the `catch` block is doing here). We should only wrap our controllers in `try...catch` blocks if we want to give context about a specific error, generic error handling is already wired in the starter code👍🏼

Reply:
Thank you for pointing `try...catch`. `asyncHanlder` actually makes the code more cleaner and also following the DRY principle.